feat: nat gateway

uny opened the pull request:
Signed-off-by: Yuki Nagai <ynagai@r.recruit.co.jp>

<!--
Add this section after we add tests and compliance
#### Submitter checklist

- [ ] Change is code complete and matches issue description.
- [ ] Change is covered by existing or new tests.
- [ ] Readme and jx-docs (https://jenkins-x.io/docs/install-setup/installing/create-cluster/eks/) are updated 
-->
#### Description
We would like to use nat gateways for private subnets.

```terraform
  private_subnet_tags = {
    "kubernetes.io/cluster/${var.cluster_name}" = "shared"
    "kubernetes.io/role/internal-elb"           = "1"
  }
```
This block is additional.
I referred [here](https://docs.aws.amazon.com/eks/latest/userguide/load-balancing.html) and [here](https://github.com/terraform-aws-modules/terraform-aws-eks/blob/7de18cd9cd882f6ad105ca375b13729537df9e68/examples/managed_node_groups/main.tf#L72-L75).
If the block should be remove in this PR, please let me know.

#### Special notes for the reviewer(s)


#### Which issue this PR fixes

fixes https://github.com/jenkins-x/terraform-aws-eks-jx/issues/113

uny commented:
/assign @garethjevans

ankitm123 commented:
Can you link this PR to an issue? @uny

uny commented:
> Can you link this PR to an issue?

Sure!
I made a new issue. https://github.com/jenkins-x/terraform-aws-eks-jx/issues/113

ankitm123 commented:
@uny overall looks good. Couple of things:
* The readme is missing a description of these variables, you can add them by running `terraform-docs`.
* Would be nice to add an example under the examples folder.

ankitm123 commented:
/ok-to-test

uny commented:
> The readme is missing a description of these variables, you can add them by running terraform-docs.
> Would be nice to add an example under the examples folder.

Thank you for your review.
For README, I ran terraform-docs and paste the result.

Also I added an example according to [here](https://github.com/terraform-aws-modules/terraform-aws-eks/blob/7de18cd9cd882f6ad105ca375b13729537df9e68/examples/managed_node_groups/main.tf).

ankitm123 commented:
In the PR description, just mention the issue number for the `fixes #` field, so that once this PR is merged, the issue gets closed.

uny commented:
> In the PR description, just mention the issue number for the fixes # field, so that once this PR is merged, the issue gets closed.

I appreciate the help.
I modified the description.

ankitm123 commented:
@uny can you rebase and run `make lint`? I am seeing some linting issues.

uny commented:
@ankitm123 
I rebased, and fixed the linting issues.
```
examples/nat/main.tf
--- old/examples/nat/main.tf
+++ new/examples/nat/main.tf
@@ -1,7 +1,7 @@
 module "jx-eks" {
-    source                    = "jenkins-x/eks-jx/aws"
-    vault_user                = var.vault_user
-    cluster_in_private_subnet = true
-    enable_nat_gateway        = true
-    single_nat_gateway        = true
+  source                    = "jenkins-x/eks-jx/aws"
+  vault_user                = var.vault_user
+  cluster_in_private_subnet = true
+  enable_nat_gateway        = true
+  single_nat_gateway        = true
 }
make: *** [fmt] Error 3
```

ankitm123 commented:
/lgtm

ankitm123 commented:
:tada: This PR is included in version 1.4.0 :tada:

The release is available on [GitHub release](https://github.com/jenkins-x/terraform-aws-eks-jx/releases/tag/v1.4.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
